packages/block-library/src/footnotes/index.php

```diff
@@ -78,3 +78,129 @@ function register_block_core_footnotes() {
 	);
 }
 add_action( 'init', 'register_block_core_footnotes' );
+
+add_action(
+	'wp_after_insert_post',
+	/**
+	 * Saves the footnotes meta value to the revision.
+	 *
+	 * @param int $revision_id The revision ID.
+	 */
+	function( $revision_id ) {
+		$post_id = wp_is_post_revision( $revision_id );
+
+		if ( $post_id ) {
+			$footnotes = get_post_meta( $post_id, 'footnotes', true );
+
+			if ( $footnotes ) {
+				// Can't use update_post_meta() because it doesn't allow revisions.
+				update_metadata( 'post', $revision_id, 'footnotes', $footnotes );
+			}
+		}
+	}
+);
+
+add_action(
+	'_wp_put_post_revision',
+	/**
+	 * Keeps track of the revision ID for "rest_after_insert_{$post_type}".
+	 *
+	 * @param int $revision_id The revision ID.
+	 */
+	function( $revision_id ) {
+		global $_gutenberg_revision_id;
+		$_gutenberg_revision_id = $revision_id;
+	}
+);
+
+foreach ( array( 'post', 'page' ) as $post_type ) {
+	add_action(
+		"rest_after_insert_{$post_type}",
+		/**
+		 * This is a specific fix for the REST API. The REST API doesn't update
+		 * the post and post meta in one go (through `meta_input`). While it
+		 * does fix the `wp_after_insert_post` hook to be called correctly after
+		 * updating meta, it does NOT fix hooks such as post_updated and
+		 * save_post, which are normally also fired after post meta is updated
+		 * in `wp_insert_post()`. Unfortunately, `wp_save_post_revision` is
+		 * added to the `post_updated` action, which means the meta is not
+		 * available at the time, so we have to add it afterwards through the
+		 * `"rest_after_insert_{$post_type}"` action.
+		 *
+		 * @param WP_Post $post The post object.
+		 */
+		function( $post ) {
+			global $_gutenberg_revision_id;
+
+			if ( $_gutenberg_revision_id ) {
+				$revision = get_post( $_gutenberg_revision_id );
+				$post_id  = $revision->post_parent;
```

Comment:
This line is causing PHP test failures on the [core backport](https://github.com/WordPress/wordpress-develop/actions/runs/5640501045/job/15277190255?pr=4892). Not 100% sure why that's happening but might be good to check if the revision exists/is an object before trying to access one of its properties?

Reply:
https://github.com/WordPress/gutenberg/pull/52879 fixes it

Reply:
I think @tellthemachines already created PR for this https://github.com/WordPress/gutenberg/pull/52870